src/status_im2/contexts/shell/activity_center/events.cljs

```diff
@@ -156,14 +156,16 @@
                         :on-error   [:activity-center/process-notification-failure nil
                                      :notification/mark-all-as-read]}]})))
 
-(rf/defn mark-all-as-read-success
-  {:events [:activity-center.notifications/mark-all-as-read-success]}
-  [{:keys [db]}]
+(defn mark-all-as-read-success
+  [{:keys [db]} [{:keys [chats] :as _messenger-response}]]
   {:db (-> (reduce (fn [acc notification-type]
                      (assoc-in acc [:activity-center :unread-counts-by-type notification-type] 0))
                    db
                    types/all-supported)
-           (update :activity-center dissoc :mark-all-as-read-undoable-till))})
+           (update :activity-center dissoc :mark-all-as-read-undoable-till))
+   :fx (map #(vector :dispatch [:chats-list/load-chat-success %]) chats)})
```

Comment:
@qfrank The best place we have IMO for isolating the responsibility of dealing with transformations from RPC responses and performing any other kind of transformation for these domain entities is `status-im.data-store`. We often use namespaces there to rename keywords, but we can go much further. Code in that directory is 100% decoupled from re-frame, simple, immutable and testable functions for handling domain entities. It's in the old directory, we should move it someday/soon. Some guidelines could be helpful in the future too.

Moving domain code outside events is also helpful for unit testing them, since there's a strong _culture_ to not test events in the mobile team.

The other advantage of writing domain code outside event namespaces is that you won't get cyclic dependency errors as easily. Same goes for subscriptions, but at least for subs we kind of test them nowadays. I would love if re-frame was used only for what it does best. Who knows, re-frame could be replaced in the future in certain parts of the app, things evolve.

Reply:
As we already have `data-store.chats/<-rpc` and the rest we need is just updating `db` , so i didn't add extra changes into `data-store` , pls let know if you're happy with the latest change 🙂 @ilmotta

Reply:
Neat @qfrank, thanks for working on this improvement. I left a separate comment as another suggestion if you like it.